src/org/project/Product.java

```diff
@@ -0,0 +1,46 @@
+package org.project;
+
+import java.math.BigDecimal;
+
+public abstract class Product {
+    private int id;
```

Comment:
w srodowisku produkcyjnym raczej nie uzywamy inta do id, czesciej jest to long, ale najczesciej w porzadnych projektach uzywamy w dzisiejszych czasach UUID https://www.baeldung.com/java-uuid, generujac za pomoca metody statycznej randomUUID